ui/packages/atlasmap/src/Atlasmap/utils/field.ts

```diff
@@ -85,100 +85,126 @@ export function getConstantTypeIndex(constVal: string): number {
 
 export function createProperty(
   propName: string,
-  propValue: string,
   propType: string,
   propScope: string,
+  isSource: boolean,
 ): void {
   const cfg = ConfigModel.getConfig();
-  let field = cfg.propertyDoc.getField(propName);
+  let field = isSource
+    ? cfg.sourcePropertyDoc.getField(propName)
+    : cfg.targetPropertyDoc.getField(propName);
   if (!field) {
     field = new Field();
   }
   field.name = propName;
-  field.value = propValue;
   field.type = propType;
   field.scope = propScope;
-  field.docDef = cfg.propertyDoc;
   field.userCreated = true;
-  cfg.propertyDoc.addField(field);
+
+  if (isSource) {
+    field.docDef = cfg.sourcePropertyDoc;
+    cfg.sourcePropertyDoc.addField(field);
+  } else {
+    field.docDef = cfg.targetPropertyDoc;
+    cfg.targetPropertyDoc.addField(field);
+  }
   cfg.mappingService.notifyMappingUpdated();
 }
 
-export function deleteProperty(propName: string): void {
+export function deleteProperty(propName: string, isSource: boolean): void {
```

Comment:
Same suggestion as above applies, if you make the document to be operated on as an argument to this, you avoid having to check if it's a source/target in all these different places.